pkg/pki/x509/x509_test.go

```diff
@@ -303,4 +310,9 @@ func TestPublicKeyWithCertChain(t *testing.T) {
 	if err == nil || !strings.Contains(err.Error(), "too many certificates specified in PEM block") {
 		t.Fatalf("expected error with long certificate chain, got %v", err)
 	}
+
+	// Verify public key with trailing newline is parsed OK
+	if _, err = NewPublicKey(strings.NewReader(pubWithTrailingNewLine)); err != nil {
```

Comment:
sorry for the extra ask for the test: but could you test this plus the same pub without the trailing new line both return the same canonicalvalue?